### PR TITLE
Fix code generator

### PR DESCRIPTION
**Pull request by JohnTortugo**
Relates to: #4728 

As discussed offline, this catches the one-after-the-end page that the paging mechanism is trying to fetch. Tested locally and the code gets generated and build correctly.

### CODE REVIEW

File: src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CodeGenerator/Languages/csharp/MethodGroup.hb [hunk 1]
```diff
@@ -85,16 +85,25 @@ namespace {{pascalCaseNs Namespace}}
...
                 while (true)
                 {
-                    var _page = await {{Name}}PageAsync(
-                        {{#each FormalParameters}}
-                        {{camelCase Name}},
-                        {{/each}}
-                        cancellationToken
-                    ).ConfigureAwait(false);
-                    if (_page.Values.Count < 1)
+                    Page<{{typeRef ResponseType.BaseType}}> _page = null;
```
Comment (alexperovich) [inline src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CodeGenerator/Languages/csharp/MethodGroup.hb]: Why not return this inside the try/catch and not have to define it here.

File: src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CodeGenerator/Languages/csharp/MethodGroup.hb [hunk 2]
```diff
@@ -85,16 +85,25 @@ namespace {{pascalCaseNs Namespace}}
...
+                            {{camelCase Name}},
+                            {{/each}}
+                            cancellationToken
+                        ).ConfigureAwait(false);
+                        if (_page.Values.Count < 1)
+                        {
+                            yield break;
+                        }                   
+                    }
+                    catch (RestApiException)
```
Comment (alexperovich) [inline src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CodeGenerator/Languages/csharp/MethodGroup.hb]: This should only catch 404 exceptions, anything else should be let through.

File: src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CodeGenerator/Languages/csharp/MethodGroup.hb (hunk 1 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CodeGenerator/Languages/csharp/MethodGroup.hb]: AFAIU I can't yield return inside a try-catch. https://stackoverflow.com/questions/346365/why-cant-yield-return-appear-inside-a-try-block-with-a-catch

Comment (alexperovich) [reply src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CodeGenerator/Languages/csharp/MethodGroup.hb]: Interesting. I guess they implemented the await-in-catch stuff but didn't do yield return.

File: src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CodeGenerator/Languages/csharp/MethodGroup.hb (hunk 2 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.SwaggerGenerator/Microsoft.DotNet.SwaggerGenerator.CodeGenerator/Languages/csharp/MethodGroup.hb]: Done

**Pull request merged**